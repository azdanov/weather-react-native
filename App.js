import React from 'react'
import {
  ActivityIndicator,
  KeyboardAvoidingView,
  Platform,
  StyleSheet,
  Text,
  ImageBackground,
  View,
} from 'react-native'

import { fetchWeather, getImage } from './src/utils'
import Search from './src/components/Search'

export default class App extends React.Component {
  state = {
    error: false,
    loading: false,
    location: '',
    temperature: 0,
    weather: '',
  }

  componentDidMount() {
    this.handleUpdateLocation('Helsinki')
  }

  handleUpdateLocation = async city => {
    if (!city) return

    this.setState({ loading: true }, async () => {
      try {
        const { location, temperature, weather } = await fetchWeather(city)

        this.setState({
          error: false,
          loading: false,
          location,
          temperature,
          weather,
        })
      } catch (error) {
        this.setState({
          error: true,
          loading: false,
        })
      }
    })
  }

  render() {
    const { error, loading, location, temperature, weather } = this.state

    return (
      <KeyboardAvoidingView style={styles.containerMain} behavior="padding">
        <ImageBackground
          imageStyle={styles.image}
          source={getImage(weather)}
          style={styles.containerImage}
        >
          <View style={styles.containerDetails}>
            <ActivityIndicator animating={loading} color="white" size="large" />

            {!loading && (
              <View>
                {error ? (
                  <Text style={[styles.textBase, styles.textSmall]}>
                    Could not load weather, please try a different city.
                  </Text>
                ) : (
                  <View>
                    <Text style={[styles.textBase, styles.textLarge]}>{location}</Text>
                    <Text style={[styles.textBase, styles.textSmall]}>{weather}</Text>
                    <Text style={[styles.textBase, styles.textLarge]}>
                      {`${Math.round(temperature)}Â°`}
                    </Text>
                  </View>
                )}

                <Search
                  placeholder="Enter a city"
                  onSubmit={this.handleUpdateLocation}
                />
              </View>
            )}
          </View>
        </ImageBackground>
      </KeyboardAvoidingView>
    )
  }
}

const styles = StyleSheet.create({
  containerDetails: {
    backgroundColor: 'hsla(0, 0%, 0%, 0.2)',
    flex: 1,
    justifyContent: 'center',
    paddingHorizontal: 20,
  },
  containerImage: { flex: 1 },
  containerMain: { backgroundColor: '#334E68', flex: 1 },
  image: { flex: 1, height: null, resizeMode: 'cover', width: null },
  textBase: {
    color: '#F0F4F8',
    textAlign: 'center',
    ...Platform.select({
      android: { fontFamily: 'Roboto' },
      ios: { fontFamily: 'AvenirNext-Regular' },
    }),
  },
  textLarge: { fontSize: 48 },
  textSmall: { fontSize: 18 },
})
