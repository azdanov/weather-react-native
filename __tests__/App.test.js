import React from 'react'
import { render } from 'react-native-testing-library'
import App from '../App'

test('should render', () => {
  const { toJSON } = render(<App />)

  expect(toJSON()).toMatchSnapshot()
})
