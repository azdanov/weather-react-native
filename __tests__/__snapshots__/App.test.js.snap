// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#334E68",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Image
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          Object {
            "flex": 1,
            "height": null,
            "resizeMode": "cover",
            "width": null,
          },
        ]
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "hsla(0, 0%, 0%, 0.2)",
          "flex": 1,
          "justifyContent": "center",
          "paddingHorizontal": 20,
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="white"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </View>
</View>
`;
